workers/social/lib/social/models/foreignauth.coffee

```diff
@@ -0,0 +1,131 @@
+jraphical = require 'jraphical'
+KodingError = require '../error'
+{ parseClient } = require './utils'
+
+module.exports = class JForeignAuth extends jraphical.Module
+
+  { secure, ObjectId, signature } = require 'bongo'
+
+  @share()
+
+  @set
+    indexes       :
+      # WARNING! ~ GG
+      # to make this working properly we need a compound index here and since
+      # bongo is not supporting them we need to manually define following:
+      #
+      #   - { username: 1, group: 1, provider: 1, foreignId: 1 } (unique)
+      #
+      provider    : 'sparse'
```

Comment:
I kept it there for development versions since we can validate those indexes for development purposes and currently compound index is just for protection since create flow doesn't allow multiple foreign auths for same user on same team with same provider.